<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="MicroBioMap">
<title>Using the MicroBioMap package • MicroBioMap</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Using the MicroBioMap package">
<meta property="og:description" content="MicroBioMap">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MicroBioMap</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.13</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/overview.html">Using the MicroBioMap package</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="http://github.com/seandavi/MicroBioMap">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Using the MicroBioMap package</h1>
            
      
      
      <div class="d-none name"><code>overview.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>The human microbiome, particularly in the large intestine, is gaining
attention for its role in health and disease, including conditions like
colorectal cancer and inflammatory bowel disease. Various factors,
including genetics and ethnicity, influence the human microbiome, often
tied to geographic regions. As a result, individuals from different
global regions tend to have distinct gut microbiomes. While research has
shown significant variation in the gut microbiome across countries and
social groups, much of the microbiome literature has focused on a
limited range of subjects, raising questions about the generalizability
of these findings.</p>
<p>To address these limitations, researchers have turned to publicly
available microbiome data, revealing an overrepresentation of
high-income countries like the United States and Finland in major
microbiome research repositories. However, significant variations from
the Western baseline are observed when other communities are studied.
This suggests that the current understanding of the microbiome may be
limited due to geographic gaps in research.</p>
<p>To bridge this gap and enhance the field’s bioinformatic
capabilities, the Human Microbiome Compendium is introduced. It
comprises over 168,000 human gut microbiome samples from 68 countries,
processed using advanced tools and made accessible in various formats,
including the MicroBioMap R package and a dedicated website. This
dataset is used to analyze global patterns in microbiome composition and
identify areas where our knowledge of the human gut microbiome is
lacking.</p>
<div class="section level3">
<h3 id="primary-data-processing">Primary data processing<a class="anchor" aria-label="anchor" href="#primary-data-processing"></a>
</h3>
<p>Amplicon processing was conducted according to the following
criteria: If the quantity of forward read files corresponded with the
number of reverse read files, we proceeded with paired-end sequencing
analysis. However, in cases of a mismatch or the absence of reverse
reads, the project was treated as single-ended data. In both scenarios,
we employed DADA2 version 1.14.0 for data processing (B. J. Callahan et
al. 2016). We utilized broad parameter settings intended to maximize the
inclusion of samples while excluding low-quality data: Specifically, we
did not apply fixed trimming from either end or impose a maximum read
length. Reads shorter than 20 nucleotides, those containing ambiguous
(“N”) base calls, and reads aligning to the phiX genome (typically
present as a control in Illumina sequencing runs) were eliminated.
Additionally, quality-based truncation of reads was disabled. Paired-end
reads were merged with a minimum overlap of 20 bases. In certain
instances, the merging process failed, resulting in minimal or no
merging of forward and reverse reads, often associated with sequencing
strategies featuring non-overlapping reads. For studies where less than
50 percent of forward reads merged successfully, we opted to discard the
reverse reads rather than concatenate them. This decision aimed to
prevent failures in merging due to low-quality calls or discrepancies
between forward and reverse read files. In such cases, the reverse reads
were excluded, and the projects were reprocessed as single-ended data.
When the number of forward reads did not align with the number of
reverse reads within a sample, an attempt was made to employ DADA2 to
identify the sequence identifier field in the FASTQ file, facilitating
the alignment of salvageable samples. In cases where this attempt proved
unsuccessful, the reverse reads were removed, and the data was
reprocessed as single-ended. Taxonomic assignment was carried out by
DADA2, utilizing the SILVA database release 138.1 as a reference (Quast
et al. 2013; McLaren and Callahan 2021).</p>
</div>
</div>
<div class="section level2">
<h2 id="getting-started">Getting started<a class="anchor" aria-label="anchor" href="#getting-started"></a>
</h2>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">BiocManager</a></span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"seandavi/MicroBioMap"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="basic-usage">Basic usage<a class="anchor" aria-label="anchor" href="#basic-usage"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://seandavi.github.io/MicroBioMap/">MicroBioMap</a></span><span class="op">)</span></span>
<span><span class="co"># this operation requires about 4GB of RAM</span></span>
<span><span class="va">cpd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getCompendium.html">getCompendium</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cpd</span></span></code></pre></div>
<pre><code><span><span class="co">## class: TreeSummarizedExperiment </span></span>
<span><span class="co">## dim: 4018 168464 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(4018):</span></span>
<span><span class="co">##   Bacteria.Firmicutes.Clostridia.Oscillospirales.Ruminococcaceae.Subdoligranulum</span></span>
<span><span class="co">##   Bacteria.Proteobacteria.Gammaproteobacteria.Burkholderiales.Alcaligenaceae.Achromobacter</span></span>
<span><span class="co">##   ...</span></span>
<span><span class="co">##   Bacteria.Proteobacteria.Alphaproteobacteria.Rhodobacterales.Rhodobacteraceae.Poseidonocella</span></span>
<span><span class="co">##   Bacteria.CK-2C2-2.NA.NA.NA.NA</span></span>
<span><span class="co">## rowData names(8): kingdom phylum ... species strain</span></span>
<span><span class="co">## colnames(168464): PRJDB10485_DRR243823 PRJDB10485_DRR243824 ...</span></span>
<span><span class="co">##   PRJNA764337_SRR15959628 PRJNA764337_SRR15959674</span></span>
<span><span class="co">## colData names(11): srs project ... iso region</span></span>
<span><span class="co">## reducedDimNames(0):</span></span>
<span><span class="co">## mainExpName: NULL</span></span>
<span><span class="co">## altExpNames(0):</span></span>
<span><span class="co">## rowLinks: NULL</span></span>
<span><span class="co">## rowTree: NULL</span></span>
<span><span class="co">## colLinks: NULL</span></span>
<span><span class="co">## colTree: NULL</span></span></code></pre>
<div class="section level3">
<h3 id="sample-metadata">Sample metadata<a class="anchor" aria-label="anchor" href="#sample-metadata"></a>
</h3>
<p>The <code>colData</code> slot of the <code>cpd</code> object contains
the information about samples. We can examine the names of the available
metadata:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">cpd</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "srs"              "project"          "srr"              "library_strategy"</span></span>
<span><span class="co">##  [5] "library_source"   "pubdate"          "total_bases"      "instrument"      </span></span>
<span><span class="co">##  [9] "geo_loc_name"     "iso"              "region"</span></span></code></pre>
<p>We can look at the samples by available regions:</p>
<ul>
<li>
<p>Region</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">cpd</span><span class="op">)</span><span class="op">$</span><span class="va">region</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##            Australia/New Zealand        Central and Southern Asia </span></span>
<span><span class="co">##                             3916                             5602 </span></span>
<span><span class="co">##   Eastern and South-Eastern Asia      Europe and Northern America </span></span>
<span><span class="co">##                            18461                           103564 </span></span>
<span><span class="co">##  Latin America and the Caribbean Northern Africa and Western Asia </span></span>
<span><span class="co">##                             1215                             1557 </span></span>
<span><span class="co">##                          Oceania               Sub-Saharan Africa </span></span>
<span><span class="co">##                                4                             5953 </span></span>
<span><span class="co">##                          unknown </span></span>
<span><span class="co">##                            28192</span></span></code></pre>
</li>
<li>
<p>Library strategy</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">cpd</span><span class="op">)</span><span class="op">$</span><span class="va">library_strategy</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## AMPLICON </span></span>
<span><span class="co">##   168464</span></span></code></pre>
</li>
<li>
<p>Instrument</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">cpd</span><span class="op">)</span><span class="op">$</span><span class="va">instrument</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##                454 GS   Illumina HiSeq 1500   Illumina HiSeq 2000 </span></span>
<span><span class="co">##                  8790                   793                  2447 </span></span>
<span><span class="co">##   Illumina HiSeq 2500   Illumina HiSeq 3000   Illumina HiSeq 4000 </span></span>
<span><span class="co">##                 10975                   701                  2651 </span></span>
<span><span class="co">##      Illumina MiniSeq        Illumina MiSeq Illumina NovaSeq 6000 </span></span>
<span><span class="co">##                   438                134353                   920 </span></span>
<span><span class="co">##           NextSeq 500           NextSeq 550           unspecified </span></span>
<span><span class="co">##                   820                   315                  5261</span></span></code></pre>
</li>
<li>
<p>Country</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">cpd</span><span class="op">)</span><span class="op">$</span><span class="va">geo_loc_name</span><span class="op">)</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##        usa:new york                                     usa             missing </span></span>
<span><span class="co">##               14142               12880               12510                5532 </span></span>
<span><span class="co">## denmark: copenhagen      not applicable </span></span>
<span><span class="co">##                5289                4317</span></span></code></pre>
</li>
</ul>
</div>
<div class="section level3">
<h3 id="microbe-row-metadata">Microbe (row) metadata<a class="anchor" aria-label="anchor" href="#microbe-row-metadata"></a>
</h3>
<p>Each row in <code>cpd</code> represents one microbe. The
<code>rowData</code> contains a data.frame- like set of metadata with
columns <code>kingdom</code>, <code>phylum</code>, etc.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">cpd</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 6 rows and 8 columns</span></span>
<span><span class="co">##                                                                                                          kingdom</span></span>
<span><span class="co">##                                                                                                      &lt;character&gt;</span></span>
<span><span class="co">## Bacteria.Firmicutes.Clostridia.Oscillospirales.Ruminococcaceae.Subdoligranulum                          Bacteria</span></span>
<span><span class="co">## Bacteria.Proteobacteria.Gammaproteobacteria.Burkholderiales.Alcaligenaceae.Achromobacter                Bacteria</span></span>
<span><span class="co">## Bacteria.Proteobacteria.Gammaproteobacteria.Enterobacterales.Enterobacteriaceae.Escherichia-Shigella    Bacteria</span></span>
<span><span class="co">## Bacteria.Firmicutes.Clostridia.Oscillospirales.Ruminococcaceae.Faecalibacterium                         Bacteria</span></span>
<span><span class="co">## Bacteria.Proteobacteria.Gammaproteobacteria.Xanthomonadales.Xanthomonadaceae.Stenotrophomonas           Bacteria</span></span>
<span><span class="co">## Bacteria.Actinobacteriota.Coriobacteriia.Coriobacteriales.Coriobacteriaceae.Collinsella                 Bacteria</span></span>
<span><span class="co">##                                                                                                                phylum</span></span>
<span><span class="co">##                                                                                                           &lt;character&gt;</span></span>
<span><span class="co">## Bacteria.Firmicutes.Clostridia.Oscillospirales.Ruminococcaceae.Subdoligranulum                             Firmicutes</span></span>
<span><span class="co">## Bacteria.Proteobacteria.Gammaproteobacteria.Burkholderiales.Alcaligenaceae.Achromobacter               Proteobacteria</span></span>
<span><span class="co">## Bacteria.Proteobacteria.Gammaproteobacteria.Enterobacterales.Enterobacteriaceae.Escherichia-Shigella   Proteobacteria</span></span>
<span><span class="co">## Bacteria.Firmicutes.Clostridia.Oscillospirales.Ruminococcaceae.Faecalibacterium                            Firmicutes</span></span>
<span><span class="co">## Bacteria.Proteobacteria.Gammaproteobacteria.Xanthomonadales.Xanthomonadaceae.Stenotrophomonas          Proteobacteria</span></span>
<span><span class="co">## Bacteria.Actinobacteriota.Coriobacteriia.Coriobacteriales.Coriobacteriaceae.Collinsella              Actinobacteriota</span></span>
<span><span class="co">##                                                                                                                    class</span></span>
<span><span class="co">##                                                                                                              &lt;character&gt;</span></span>
<span><span class="co">## Bacteria.Firmicutes.Clostridia.Oscillospirales.Ruminococcaceae.Subdoligranulum                                Clostridia</span></span>
<span><span class="co">## Bacteria.Proteobacteria.Gammaproteobacteria.Burkholderiales.Alcaligenaceae.Achromobacter             Gammaproteobacteria</span></span>
<span><span class="co">## Bacteria.Proteobacteria.Gammaproteobacteria.Enterobacterales.Enterobacteriaceae.Escherichia-Shigella Gammaproteobacteria</span></span>
<span><span class="co">## Bacteria.Firmicutes.Clostridia.Oscillospirales.Ruminococcaceae.Faecalibacterium                               Clostridia</span></span>
<span><span class="co">## Bacteria.Proteobacteria.Gammaproteobacteria.Xanthomonadales.Xanthomonadaceae.Stenotrophomonas        Gammaproteobacteria</span></span>
<span><span class="co">## Bacteria.Actinobacteriota.Coriobacteriia.Coriobacteriales.Coriobacteriaceae.Collinsella                   Coriobacteriia</span></span>
<span><span class="co">##                                                                                                                 order</span></span>
<span><span class="co">##                                                                                                           &lt;character&gt;</span></span>
<span><span class="co">## Bacteria.Firmicutes.Clostridia.Oscillospirales.Ruminococcaceae.Subdoligranulum                        Oscillospirales</span></span>
<span><span class="co">## Bacteria.Proteobacteria.Gammaproteobacteria.Burkholderiales.Alcaligenaceae.Achromobacter              Burkholderiales</span></span>
<span><span class="co">## Bacteria.Proteobacteria.Gammaproteobacteria.Enterobacterales.Enterobacteriaceae.Escherichia-Shigella Enterobacterales</span></span>
<span><span class="co">## Bacteria.Firmicutes.Clostridia.Oscillospirales.Ruminococcaceae.Faecalibacterium                       Oscillospirales</span></span>
<span><span class="co">## Bacteria.Proteobacteria.Gammaproteobacteria.Xanthomonadales.Xanthomonadaceae.Stenotrophomonas         Xanthomonadales</span></span>
<span><span class="co">## Bacteria.Actinobacteriota.Coriobacteriia.Coriobacteriales.Coriobacteriaceae.Collinsella              Coriobacteriales</span></span>
<span><span class="co">##                                                                                                                  family</span></span>
<span><span class="co">##                                                                                                             &lt;character&gt;</span></span>
<span><span class="co">## Bacteria.Firmicutes.Clostridia.Oscillospirales.Ruminococcaceae.Subdoligranulum                          Ruminococcaceae</span></span>
<span><span class="co">## Bacteria.Proteobacteria.Gammaproteobacteria.Burkholderiales.Alcaligenaceae.Achromobacter                 Alcaligenaceae</span></span>
<span><span class="co">## Bacteria.Proteobacteria.Gammaproteobacteria.Enterobacterales.Enterobacteriaceae.Escherichia-Shigella Enterobacteriaceae</span></span>
<span><span class="co">## Bacteria.Firmicutes.Clostridia.Oscillospirales.Ruminococcaceae.Faecalibacterium                         Ruminococcaceae</span></span>
<span><span class="co">## Bacteria.Proteobacteria.Gammaproteobacteria.Xanthomonadales.Xanthomonadaceae.Stenotrophomonas          Xanthomonadaceae</span></span>
<span><span class="co">## Bacteria.Actinobacteriota.Coriobacteriia.Coriobacteriales.Coriobacteriaceae.Collinsella               Coriobacteriaceae</span></span>
<span><span class="co">##                                                                                                                     genus</span></span>
<span><span class="co">##                                                                                                               &lt;character&gt;</span></span>
<span><span class="co">## Bacteria.Firmicutes.Clostridia.Oscillospirales.Ruminococcaceae.Subdoligranulum                            Subdoligranulum</span></span>
<span><span class="co">## Bacteria.Proteobacteria.Gammaproteobacteria.Burkholderiales.Alcaligenaceae.Achromobacter                    Achromobacter</span></span>
<span><span class="co">## Bacteria.Proteobacteria.Gammaproteobacteria.Enterobacterales.Enterobacteriaceae.Escherichia-Shigella Escherichia-Shigella</span></span>
<span><span class="co">## Bacteria.Firmicutes.Clostridia.Oscillospirales.Ruminococcaceae.Faecalibacterium                          Faecalibacterium</span></span>
<span><span class="co">## Bacteria.Proteobacteria.Gammaproteobacteria.Xanthomonadales.Xanthomonadaceae.Stenotrophomonas            Stenotrophomonas</span></span>
<span><span class="co">## Bacteria.Actinobacteriota.Coriobacteriia.Coriobacteriales.Coriobacteriaceae.Collinsella                       Collinsella</span></span>
<span><span class="co">##                                                                                                          species</span></span>
<span><span class="co">##                                                                                                      &lt;character&gt;</span></span>
<span><span class="co">## Bacteria.Firmicutes.Clostridia.Oscillospirales.Ruminococcaceae.Subdoligranulum                                NA</span></span>
<span><span class="co">## Bacteria.Proteobacteria.Gammaproteobacteria.Burkholderiales.Alcaligenaceae.Achromobacter                      NA</span></span>
<span><span class="co">## Bacteria.Proteobacteria.Gammaproteobacteria.Enterobacterales.Enterobacteriaceae.Escherichia-Shigella          NA</span></span>
<span><span class="co">## Bacteria.Firmicutes.Clostridia.Oscillospirales.Ruminococcaceae.Faecalibacterium                               NA</span></span>
<span><span class="co">## Bacteria.Proteobacteria.Gammaproteobacteria.Xanthomonadales.Xanthomonadaceae.Stenotrophomonas                 NA</span></span>
<span><span class="co">## Bacteria.Actinobacteriota.Coriobacteriia.Coriobacteriales.Coriobacteriaceae.Collinsella                       NA</span></span>
<span><span class="co">##                                                                                                           strain</span></span>
<span><span class="co">##                                                                                                      &lt;character&gt;</span></span>
<span><span class="co">## Bacteria.Firmicutes.Clostridia.Oscillospirales.Ruminococcaceae.Subdoligranulum                                NA</span></span>
<span><span class="co">## Bacteria.Proteobacteria.Gammaproteobacteria.Burkholderiales.Alcaligenaceae.Achromobacter                      NA</span></span>
<span><span class="co">## Bacteria.Proteobacteria.Gammaproteobacteria.Enterobacterales.Enterobacteriaceae.Escherichia-Shigella          NA</span></span>
<span><span class="co">## Bacteria.Firmicutes.Clostridia.Oscillospirales.Ruminococcaceae.Faecalibacterium                               NA</span></span>
<span><span class="co">## Bacteria.Proteobacteria.Gammaproteobacteria.Xanthomonadales.Xanthomonadaceae.Stenotrophomonas                 NA</span></span>
<span><span class="co">## Bacteria.Actinobacteriota.Coriobacteriia.Coriobacteriales.Coriobacteriaceae.Collinsella                       NA</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="use-cases">Use cases<a class="anchor" aria-label="anchor" href="#use-cases"></a>
</h2>
<div class="section level3">
<h3 id="samples-in-specific-bioprojects">Samples in specific Bioprojects<a class="anchor" aria-label="anchor" href="#samples-in-specific-bioprojects"></a>
</h3>
<p>To select samples that belong to specific Bioprojects, we can filter
using the “project” metadata column in the <code>colData</code>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">cpd</span><span class="op">)</span><span class="op">$</span><span class="va">project</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "PRJDB10485" "PRJDB10527" "PRJDB10528" "PRJDB10612" "PRJDB11845"</span></span>
<span><span class="co">## [6] "PRJDB11894"</span></span></code></pre>
<p>For example, create a subset of data belonging to Bioprojects:</p>
<ul>
<li>PRJDB10485</li>
<li>PRJDB10527</li>
<li>PRJDB10528</li>
<li>PRJDB10612</li>
</ul>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">project_ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PRJDB10485"</span>, <span class="st">"PRJDB10527"</span>, <span class="st">"PRJDB10528"</span>, <span class="st">"PRJDB10612"</span><span class="op">)</span></span>
<span><span class="va">cpd_sub_by_project_ids</span> <span class="op">&lt;-</span> <span class="va">cpd</span><span class="op">[</span>, <span class="fu">colData</span><span class="op">(</span><span class="va">cpd</span><span class="op">)</span><span class="op">$</span><span class="va">project</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">project_ids</span><span class="op">]</span></span>
<span><span class="va">cpd_sub_by_project_ids</span></span></code></pre></div>
<pre><code><span><span class="co">## class: TreeSummarizedExperiment </span></span>
<span><span class="co">## dim: 4018 637 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(4018):</span></span>
<span><span class="co">##   Bacteria.Firmicutes.Clostridia.Oscillospirales.Ruminococcaceae.Subdoligranulum</span></span>
<span><span class="co">##   Bacteria.Proteobacteria.Gammaproteobacteria.Burkholderiales.Alcaligenaceae.Achromobacter</span></span>
<span><span class="co">##   ...</span></span>
<span><span class="co">##   Bacteria.Proteobacteria.Alphaproteobacteria.Rhodobacterales.Rhodobacteraceae.Poseidonocella</span></span>
<span><span class="co">##   Bacteria.CK-2C2-2.NA.NA.NA.NA</span></span>
<span><span class="co">## rowData names(8): kingdom phylum ... species strain</span></span>
<span><span class="co">## colnames(637): PRJDB10485_DRR243823 PRJDB10485_DRR243824 ...</span></span>
<span><span class="co">##   PRJDB10612_DRR254611 PRJDB10612_DRR254699</span></span>
<span><span class="co">## colData names(11): srs project ... iso region</span></span>
<span><span class="co">## reducedDimNames(0):</span></span>
<span><span class="co">## mainExpName: NULL</span></span>
<span><span class="co">## altExpNames(0):</span></span>
<span><span class="co">## rowLinks: NULL</span></span>
<span><span class="co">## rowTree: NULL</span></span>
<span><span class="co">## colLinks: NULL</span></span>
<span><span class="co">## colTree: NULL</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="samples-from-a-specific-world-region">Samples from a specific world region<a class="anchor" aria-label="anchor" href="#samples-from-a-specific-world-region"></a>
</h3>
<p>To select a subset of samples from a specific world region, subset by
the “columns” of the dataset. The following code ummarized the regions
are available and the number of samples in each region.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">cpd</span><span class="op">)</span><span class="op">$</span><span class="va">region</span><span class="op">)</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##      Europe and Northern America                          unknown </span></span>
<span><span class="co">##                           103564                            28192 </span></span>
<span><span class="co">##   Eastern and South-Eastern Asia               Sub-Saharan Africa </span></span>
<span><span class="co">##                            18461                             5953 </span></span>
<span><span class="co">##        Central and Southern Asia            Australia/New Zealand </span></span>
<span><span class="co">##                             5602                             3916 </span></span>
<span><span class="co">## Northern Africa and Western Asia  Latin America and the Caribbean </span></span>
<span><span class="co">##                             1557                             1215 </span></span>
<span><span class="co">##                          Oceania </span></span>
<span><span class="co">##                                4</span></span></code></pre>
<p>Now, select only those samples from, for example, “Sub-Saharan
Africa”.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cpd_africa</span> <span class="op">&lt;-</span> <span class="va">cpd</span><span class="op">[</span>, <span class="fu">colData</span><span class="op">(</span><span class="va">cpd</span><span class="op">)</span><span class="op">$</span><span class="va">region</span> <span class="op">==</span> <span class="st">"Sub-Saharan Africa"</span><span class="op">]</span></span>
<span><span class="va">cpd_africa</span></span></code></pre></div>
<pre><code><span><span class="co">## class: TreeSummarizedExperiment </span></span>
<span><span class="co">## dim: 4018 5953 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(4018):</span></span>
<span><span class="co">##   Bacteria.Firmicutes.Clostridia.Oscillospirales.Ruminococcaceae.Subdoligranulum</span></span>
<span><span class="co">##   Bacteria.Proteobacteria.Gammaproteobacteria.Burkholderiales.Alcaligenaceae.Achromobacter</span></span>
<span><span class="co">##   ...</span></span>
<span><span class="co">##   Bacteria.Proteobacteria.Alphaproteobacteria.Rhodobacterales.Rhodobacteraceae.Poseidonocella</span></span>
<span><span class="co">##   Bacteria.CK-2C2-2.NA.NA.NA.NA</span></span>
<span><span class="co">## rowData names(8): kingdom phylum ... species strain</span></span>
<span><span class="co">## colnames(5953): PRJDB10485_DRR243823 PRJDB10485_DRR243824 ...</span></span>
<span><span class="co">##   PRJNA706727_SRR13886715 PRJNA706727_SRR13886719</span></span>
<span><span class="co">## colData names(11): srs project ... iso region</span></span>
<span><span class="co">## reducedDimNames(0):</span></span>
<span><span class="co">## mainExpName: NULL</span></span>
<span><span class="co">## altExpNames(0):</span></span>
<span><span class="co">## rowLinks: NULL</span></span>
<span><span class="co">## rowTree: NULL</span></span>
<span><span class="co">## colLinks: NULL</span></span>
<span><span class="co">## colTree: NULL</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="samples-from-a-specific-country">Samples from a specific country<a class="anchor" aria-label="anchor" href="#samples-from-a-specific-country"></a>
</h3>
<p>To select a subset of samples from a specific country, subset by the
“columns” of the dataset. The following code ummarized the countries are
available and the number of samples in each country.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://forcats.tidyverse.org/" class="external-link">forcats</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">cpd</span><span class="op">)</span><span class="op">)</span>, </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://forcats.tidyverse.org/reference/fct_inorder.html" class="external-link">fct_infreq</a></span><span class="op">(</span><span class="va">iso</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat<span class="op">=</span><span class="st">'count'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Country"</span>, y <span class="op">=</span> <span class="st">"Number of samples"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_log10</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="enrich-data-with-country-names">Enrich data with country names<a class="anchor" aria-label="anchor" href="#enrich-data-with-country-names"></a>
</h3>
<p>Using the <code>countrycode</code> package, we can convert the iso
codes to country names.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://vincentarelbundock.github.io/countrycode/" class="external-link">countrycode</a></span><span class="op">)</span></span>
<span><span class="fu">colData</span><span class="op">(</span><span class="va">cpd</span><span class="op">)</span><span class="op">$</span><span class="va">country</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://vincentarelbundock.github.io/countrycode/reference/countrycode.html" class="external-link">countrycode</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">cpd</span><span class="op">)</span><span class="op">$</span><span class="va">iso</span>, <span class="st">"iso2c"</span>, <span class="st">"country.name"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Some values were not matched unambiguously: UNKNOWN</span></span></code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># or add the iso3c code</span></span>
<span><span class="fu">colData</span><span class="op">(</span><span class="va">cpd</span><span class="op">)</span><span class="op">$</span><span class="va">iso3c</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://vincentarelbundock.github.io/countrycode/reference/countrycode.html" class="external-link">countrycode</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">cpd</span><span class="op">)</span><span class="op">$</span><span class="va">iso</span>, <span class="st">"iso2c"</span>, <span class="st">"iso3c"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Some values were not matched unambiguously: UNKNOWN</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="histogram-of-total-bases-per-sample">Histogram of total bases per sample<a class="anchor" aria-label="anchor" href="#histogram-of-total-bases-per-sample"></a>
</h3>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">cpd</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">total_bases</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html" class="external-link">geom_histogram</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-16-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="samples-filtered-by-presence-of-microbe">Samples filtered by presence of microbe<a class="anchor" aria-label="anchor" href="#samples-filtered-by-presence-of-microbe"></a>
</h3>
<p>There are over 4000 microbes represented in the compendium. Some are
fairly rare and we may want to subset the compendium to only those with
a specific bug in the sample. Here, we pick an arbitrary examplar to
show just the mechanics.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">shig_cpd_counts</span> <span class="op">=</span> <span class="fu">counts</span><span class="op">(</span><span class="va">cpd</span><span class="op">)</span><span class="op">[</span><span class="st">'Bacteria.Proteobacteria.Gammaproteobacteria.Enterobacterales.Enterobacteriaceae.Escherichia-Shigella'</span>,<span class="op">]</span></span></code></pre></div>
<p>Examine the distribution of abundance across all samples in the
compendium.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">shig_cpd_counts</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-18-1.png" width="700">
Picking an arbitrary threshold (after all, this is only an example), we
can limit samples to those with plenty of our bacteria of interest
present.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">shig_cpd</span> <span class="op">&lt;-</span> <span class="va">cpd</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">shig_cpd_counts</span><span class="op">)</span><span class="op">&gt;</span><span class="fl">2</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">shig_cpd</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1]  4018 72769</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="sessioninfo">sessionInfo<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h2>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.0 (2023-04-21)</span></span>
<span><span class="co">## Platform: aarch64-apple-darwin20 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Ventura 13.1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: America/New_York</span></span>
<span><span class="co">## tzcode source: internal</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] countrycode_1.5.0              ggplot2_3.4.2                 </span></span>
<span><span class="co">##  [3] forcats_1.0.0                  MicroBioMap_0.99.13           </span></span>
<span><span class="co">##  [5] ape_5.7-1                      TreeSummarizedExperiment_2.8.0</span></span>
<span><span class="co">##  [7] Biostrings_2.68.1              XVector_0.40.0                </span></span>
<span><span class="co">##  [9] SingleCellExperiment_1.22.0    SummarizedExperiment_1.30.2   </span></span>
<span><span class="co">## [11] Biobase_2.60.0                 GenomicRanges_1.52.0          </span></span>
<span><span class="co">## [13] GenomeInfoDb_1.36.1            IRanges_2.34.1                </span></span>
<span><span class="co">## [15] S4Vectors_0.38.1               BiocGenerics_0.46.0           </span></span>
<span><span class="co">## [17] MatrixGenerics_1.12.2          matrixStats_1.0.0             </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] tidyselect_1.2.0        farver_2.1.1            dplyr_1.1.2            </span></span>
<span><span class="co">##  [4] blob_1.2.4              filelock_1.0.2          R.utils_2.12.2         </span></span>
<span><span class="co">##  [7] bitops_1.0-7            fastmap_1.1.1           RCurl_1.98-1.12        </span></span>
<span><span class="co">## [10] lazyeval_0.2.2          BiocFileCache_2.8.0     digest_0.6.33          </span></span>
<span><span class="co">## [13] lifecycle_1.0.3         tidytree_0.4.2          RSQLite_2.3.1          </span></span>
<span><span class="co">## [16] magrittr_2.0.3          compiler_4.3.0          rlang_1.1.1            </span></span>
<span><span class="co">## [19] sass_0.4.6              tools_4.3.0             utf8_1.2.3             </span></span>
<span><span class="co">## [22] yaml_2.3.7              data.table_1.14.8       knitr_1.43             </span></span>
<span><span class="co">## [25] labeling_0.4.2          S4Arrays_1.0.4          bit_4.0.5              </span></span>
<span><span class="co">## [28] curl_5.0.1              DelayedArray_0.26.6     BiocParallel_1.34.2    </span></span>
<span><span class="co">## [31] withr_2.5.0             purrr_1.0.1             desc_1.4.2             </span></span>
<span><span class="co">## [34] R.oo_1.25.0             grid_4.3.0              fansi_1.0.4            </span></span>
<span><span class="co">## [37] colorspace_2.1-0        scales_1.2.1            cli_3.6.1              </span></span>
<span><span class="co">## [40] rmarkdown_2.23          crayon_1.5.2            ragg_1.2.5             </span></span>
<span><span class="co">## [43] treeio_1.24.1           generics_0.1.3          httr_1.4.6             </span></span>
<span><span class="co">## [46] DBI_1.1.3               cachem_1.0.8            stringr_1.5.0          </span></span>
<span><span class="co">## [49] zlibbioc_1.46.0         parallel_4.3.0          vctrs_0.6.3            </span></span>
<span><span class="co">## [52] yulab.utils_0.0.6       Matrix_1.6-0            jsonlite_1.8.7         </span></span>
<span><span class="co">## [55] bit64_4.0.5             systemfonts_1.0.4       jquerylib_0.1.4        </span></span>
<span><span class="co">## [58] tidyr_1.3.0             glue_1.6.2              pkgdown_2.0.7          </span></span>
<span><span class="co">## [61] codetools_0.2-19        gtable_0.3.3            stringi_1.7.12         </span></span>
<span><span class="co">## [64] munsell_0.5.0           tibble_3.2.1            pillar_1.9.0           </span></span>
<span><span class="co">## [67] htmltools_0.5.5         GenomeInfoDbData_1.2.10 R6_2.5.1               </span></span>
<span><span class="co">## [70] dbplyr_2.3.3            textshaping_0.3.6       rprojroot_2.0.3        </span></span>
<span><span class="co">## [73] evaluate_0.21           lattice_0.21-8          highr_0.10             </span></span>
<span><span class="co">## [76] R.methodsS3_1.8.2       memoise_2.0.1           bslib_0.5.0            </span></span>
<span><span class="co">## [79] Rcpp_1.0.11             nlme_3.1-162            xfun_0.39              </span></span>
<span><span class="co">## [82] fs_1.6.2                pkgconfig_2.0.3</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Sean Davis, Richard Abdill, Ran Blehkman, Samantha Graham, Casey Greene.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
